src/ecdsa/keys.py

```diff
@@ -1133,3 +1133,28 @@ def sign_number(self, number, entropy=None, k=None):
         assert 1 <= _k < order
         sig = self.privkey.sign(number, _k)
         return sig.r, sig.s
+
+    def ecdh_get_shared_secret(self, remote_public_key):
+        """
+        Elliptic-curve Diffie-Hellman (ECDH). A key agreement protocol.
+        Allows two parties, each having an elliptic-curve public-private key 
+        pair, to establish a shared secret over an insecure channel
+                
+        :param remote_public_key: public key of remote party (VerifyingKey)
+        :return: shared secret
+        """""
+
+        if not self.curve.curve.contains_point(
+                remote_public_key.pubkey.point.x(),
+                remote_public_key.pubkey.point.y()):
+            raise ValueError("Invalid curve in public key.")
```

Comment:
this does not check if the if the point is a valid public key on the curve, for this two checks need to be performed:
https://github.com/warner/python-ecdsa/blob/ca11bcb62ae576e0fd2140fe1fa89ab90e3c5ec1/src/ecdsa/ellipticcurve.py#L75-L78

but combined with this code:
https://github.com/warner/python-ecdsa/blob/ca11bcb62ae576e0fd2140fe1fa89ab90e3c5ec1/src/ecdsa/ecdsa.py#L117-L118

it's always performed for points in the `Public_key` object (and since it's an expensive check we really don't want to do it twice)

second, this does not check if the _curves_ match, the code should simply compare `self.curve.curve` with `remote.curve.curve`